src/workers/_quality_control.js

```diff
@@ -152,10 +152,13 @@ export function results() {
         ranges[k] = [min, max];
     }
 
+    var mat = fetchFilteredMatrix();
+
     return { 
         "data": data, 
         "ranges": ranges,
-        "thresholds": thresholds
+        "thresholds": thresholds,
+        "dims": [mat.numberOfRows(), mat.numberOfColumns()]
```

Comment:
- Why do you need `mat.numberOfRows()`? I thought you just needed the number left after QC.
- To get the number left after QC, it would be better to do something like:

```js
let remaining;
if ("matrix" in cache) {
     remaining = cache.matrix.numberOfColumns();
} else {
     remaining = 0;
     fetchDiscards().array().forEach(x => {
          if (x == 0) {
                 remaining++;
          }
      });
}
```

This avoids generating the filtered matrix until it is actually needed.